amads/time/meter/grid.py

```diff
@@ -0,0 +1,181 @@
+"""
+Settle on an appropriate granular grid of metrical tatums in response to a source and user settings.
+"""
+
+__author__ = "Mark Gotham"
+
+
+# ------------------------------------------------------------------------------
+
+import logging
+from collections import Counter
+from typing import Iterable, Union
+
+logging.basicConfig(
+    level=logging.DEBUG, format="%(asctime)s - %(levelname)s - %(message)s"
+)
+
+
+def starts_to_measure_relative_counter(starts: Iterable[float]):
+    """
+    Simple wrapper function to map an iterable (e.g., list or tuple) of floats to
+    a measure-relative Counter, such that all the keys are geq 0, and less than 1.
+
+    Examples
+    --------
+    >>> test_list = [0.0, 0.0, 0.5, 1.0, 1.5, 1.75, 2.0, 2.33, 2.667, 3]
+    >>> starts_to_measure_relative_counter(test_list)
+    Counter({0.0: 5, 0.5: 2, 0.75: 1, 0.33000000000000007: 1, 0.6669999999999998: 1})
+    """
+    for item in starts:
+        if not isinstance(item, (int, float)):
+            raise TypeError(
+                f"All items in `starts` must be numeric (int or float). Found: {type(item)}"
+            )
+
+    return Counter([(x - int(x)) for x in starts])
```

Comment:
This seems like a case where you might want to round to a set number of digits?

Reply:
Ok, with caveat as below.